### DISCUSSION TITLE
PvPGN server on Mac (private Battle.net)

### DISCUSSION THREAD

**Discussion by ChristofferSthlm** [Homebrew/discussions#4148] [Everyday usage]
Hi there!

Wondering if it's possible to install the self-hosted Battle.net-server PvPGN on Mac OS X (Silicon). I'm looking in to the commands but cannot really wrap my head around it. Seems for example like build-essential is not available through Brew? If someone with more skills can look into it, I would be thankful! :) 

Link to PvPGN with installation/compilation details:
https://github.com/pvpgn/pvpgn-server

**Comment by p-linnane** [comment]
https://docs.brew.sh/Adding-Software-to-Homebrew

**Comment by ChristofferSthlm** [comment]
Thanks, but it's not helping too much, I'm afraid. The tay PvPGN comes at the moment, it needs to be built at installation.

In Linux in general, it looks like this:

`apt-get install git install cmake make build-essential zlib1g-dev
apt-get install liblua5.1-0-dev #Lua support
apt-get install mysql-server mysql-client libmysqlclient-dev #MySQL support
cd /home
git clone https://github.com/pvpgn/pvpgn-server.git
cmake -D CMAKE_INSTALL_PREFIX=/usr/local/pvpgn -D WITH_MYSQL=true -D WITH_LUA=true ../
make
make install`

However, I really don't know anything about if **cmake**, **build-essential**, **zlib1g-dev** or **liblua5.1-0-dev** works on Mac OS, or what to use instead. Cmake seems to be working on the mac, since it's on the brew list, but build-essential is not. Here's where I need help. :)

**Reply by SMillerDev** [reply]
Homebrew can't really tell you what upstream needs to build. They could need everything in build-essentials from whichever Debian version or only a single tool for a specific version. Homebrew maintainers don't know how their build process works either.

**Reply by ChristofferSthlm** [reply]
True. Was just hoping anyone over here would know. :) Also since I would install the packages/apps with brew instead of apt-get on mac, and maybe eventually be able to add PvPGN as a software to brew, although I believe I know all to little about those things to do so.
